### PR TITLE
DwarfEH: 32 bit Linux support

### PR DESCRIPTION
**Pull request by WalterBright**
This is a work in progress.

### CODE REVIEW

**Comment by WalterBright** [comment]
Requires https://github.com/D-Programming-Language/druntime/pull/1474

**Comment by WalterBright** [comment]
This is a bit frustrating. The test case druntime/test/shared/finalizer is seg faulting after main() returns, and it's failing in gc_term(). It only happens when linking to shared libraries in release build. Anyone have any ideas?

**Comment by WalterBright** [comment]
Pull #1482 fixed it. Ready to pull.

File: src/backend/elfobj.c [hunk 1]
```diff
@@ -3522,10 +3520,10 @@ int dwarf_reftoident(int seg, targ_size_t offset, Symbol *s, targ_size_t val)
          */
         if (!s->Sdw_ref_idx)
         {
-            int dataDWref_seg = ElfObj::getsegment(".data.DW.ref.", s->Sident, SHT_PROGBITS, SHF_ALLOC|SHF_WRITE, 8);
+            int dataDWref_seg = ElfObj::getsegment(".data.DW.ref.", s->Sident, SHT_PROGBITS, SHF_ALLOC|SHF_WRITE, I64 ? 8 : 4);
```
Comment (andralex) [inline src/backend/elfobj.c]: longish line

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**